libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/HttpHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.WebApi
+{
+    public static class HttpHelper
+    {
+        public static readonly MediaTypeFormatter[] BotMessageMediaTypeFormatters = new[]
+        {
+            new JsonMediaTypeFormatter
+            {
+                SerializerSettings = MessageSerializerSettings.Create(),
+                SupportedMediaTypes =
+                {
+                    new System.Net.Http.Headers.MediaTypeHeaderValue("application/json") { CharSet = "utf-8" },
+                    new System.Net.Http.Headers.MediaTypeHeaderValue("text/json") { CharSet = "utf-8" },
+                },
+            },
+        };
+
+        public static async Task<Activity> FromRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var activity = await request.Content.ReadAsAsync<Activity>(BotMessageMediaTypeFormatters, cancellationToken).ConfigureAwait(false);
+            return activity;
+        }
+
+        public static void ToResponse(HttpRequestMessage request, HttpResponseMessage response, InvokeResponse invokeResponse)
```

Comment:
The `request` doesn't seem to be used... or necessary. Consider removing the param?

Reply:
I'm mimicing teh exact behavior of the extension request.CreateResponse() which sets the Request object onto the Response object. I realize not strictly necessary but consistent.